csharp-basic-topics/ValidateNumberIsPositiveOrNegative/ValidateNumberIsPositiveOrNegative/NumberValidation.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ValidateNumberIsPositiveOrNegative
+{
+    public static class NumberValidation
+    {
+        public static int IsPositiveOrNegativeUsingConditionalMethod(int number)
+        {
+            var result = 0;
+
+            if (number > 0)
+            {
+                result = 1;
+            }
+            else if (number < 0)
+            {
+                result = - 1;
+            }
+            else
+            {
+                result = 0;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingLeftShiftMethod(int number)
+        {
+            var result = 0;
+            // Using left shift operator to check the sign bit
+            if ((number & (1 << 31)) == 0)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = -1;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingRightShiftMethod(int number)
+        {
+            var result = 0;
+            // Using right shift operator to check the sign bit
+            if ((number >> 31) == 0)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = -1;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingMathAbsMethod(int number)
+        {
```

Comment:
Same comment for zero. Also, no need to store the result of Math.Abs in a separate variable, directly we can check in if statement.

Reply:
Updated method for case zero.